pytest_plugins/issue_handlers.py

```diff
@@ -0,0 +1,272 @@
+import inspect
+import json
+import logging
+import re
+from collections import defaultdict
+from datetime import datetime
+
+import pytest
+
+from robottelo.config import settings
+from robottelo.helpers import slugify_component
+from robottelo.utils.issue_handlers import add_workaround
+from robottelo.utils.issue_handlers import bugzilla
+from robottelo.utils.issue_handlers import is_open
+from robottelo.utils.issue_handlers import should_deselect
+from robottelo.utils.version import search_version_key
+from robottelo.utils.version import VersionEncoder
+
+LOGGER = logging.getLogger('robottelo')
+
+DEFAULT_BZ_CACHE_FILE = 'bz_cache.json'
+
+
+def pytest_addoption(parser):
+    """Adds custom options to pytest runner."""
+    parser.addoption(
+        "--bz-cache",
+        action='store_true',
+        help=f"Use an existing {DEFAULT_BZ_CACHE_FILE} instead of calling BZ API. "
+        "One will be created automatically by default, but not used by default",
+    )
+
+
+@pytest.hookimpl(trylast=True)
+def pytest_collection_modifyitems(session, items, config):
+    """Generate the issue collection (using bz cache via bugzilla issue handler util"""
+    # This collection includes pre-processed `is_open` status for each issue
+    # generate_issue_collection will save a file, set by --bz-cache value, on each run.
+    pytest.issue_data = generate_issue_collection(items, config)
+
+    # No item deselect, just adding skipif markers
+    for item in items:
+        skip_if_open = item.get_closest_marker('skip_if_open')
+        if skip_if_open:
+            # marker must have `BZ:123456` as argument.
+            issue = skip_if_open.kwargs.get('reason') or skip_if_open.args[0]
+            item.add_marker(pytest.mark.skipif(is_open(issue), reason=issue))
+
+
+IS_OPEN = re.compile(
+    # To match `if is_open('BZ:123456'):`
+    r"\s*if\sis_open\(\S(?P<src>\D{2})\s*:\s*(?P<num>\d*)\S\)\d*"
+)
+
+NOT_IS_OPEN = re.compile(
+    # To match `if not is_open('BZ:123456'):`
+    r"\s*if\snot\sis_open\(\S(?P<src>\D{2})\s*:\s*(?P<num>\d*)\S\)\d*"
+)
+
+COMPONENT = re.compile(
+    # To match :CaseComponent: FooBar
+    r"\s*:CaseComponent:\s*(?P<component>\S*)",
+    re.IGNORECASE,
+)
+
+IMPORTANCE = re.compile(
+    # To match :CaseImportance: Critical
+    r"\s*:CaseImportance:\s*(?P<importance>\S*)",
+    re.IGNORECASE,
+)
+
+BZ = re.compile(
+    # To match :BZ: 123456, 456789
+    r"\s*:BZ:\s*(?P<bz>.*\S*)",
+    re.IGNORECASE,
+)
+
+
+def generate_issue_collection(items, config):  # pragma: no cover
```

Comment:
Agreed on it being a bit too long, but I would like to avoid re-factoring the contents into pieces within this PR, given the scope. Would you agree its acceptable as is (copied and modified from original location), and that we can break it up further as we have the need?

Reply:
@mshriver Agree, lets open another PR for it.